pkg/river/internal/value/decode.go

```diff
@@ -126,6 +142,14 @@ func (d *decoder) decode(val Value, into reflect.Value) error {
 		return err
 	}
 
+	if into.CanAddr() {
+		if into.Addr().Type().Implements(goRiverDefaulter) {
+			into.Addr().Interface().(Defaulter).SetToDefault()
+		}
+	} else if into.Type().Implements(goRiverDefaulter) {
+		into.Interface().(Defaulter).SetToDefault()
+	}
```

Comment:
nit to clean up the depth:

```suggestion
	if into.CanAddr() && into.Addr().Type().Implements(goRiverDefaulter) {
		into.Addr().Interface().(Defaulter).SetToDefault()
	} else if into.Type().Implements(goRiverDefaulter) {
		into.Interface().(Defaulter).SetToDefault()
	}
```